internal/resolver/resolver.go

```diff
@@ -52,6 +54,19 @@ func New(chartpath, cachepath string, registryClient *registry.Client) *Resolver
 	}
 }
 
+// DependencyKey creates a unique key for a dependency that includes name, repository, and index
+// to handle cases where multiple dependencies have the same name but different repositories
+func DependencyKey(name, repository string, index int) string {
+	// Use a combination of name, repository hash, and index to ensure uniqueness
+	repoHash := ""
+	if repository != "" {
+		hasher := sha256.New()
+		hasher.Write([]byte(repository))
+		repoHash = hex.EncodeToString(hasher.Sum(nil))[:8] // Use first 8 chars for brevity
```

Comment:
I think 8 is good enough with the amount of chance and usage of the same name in `Chart.yaml`.

But I'm not sure we need the dependency index from the key. name+normalized-repository should be sufficient for map uniqueness. If the same name+repo appears twice, both entries map to the same repo, so overwriting is harmless at this mapping layer and removes index-coupling. What do you think ?